### ISSUE TITLE
Can't retrieve temperature

### ISSUE THREAD

**Issue by Pgr73** [flobz/psa_car_controller#16]
Hi @flobz, VERY nice piece of work you have done here with this PSA car controller interface ! Thank you very much for having shared it !

This is probably not a bug in your code and more a question/improvement suggestion.

I'd like to retrieve the temperature which, according to the PSA API documentation [https://developer.groupe-psa.io/webapi/b2c/quickstart/connect/#article] is accessible through two possible API paths : `/vehicle/{id}/telemetry` and the `/vehicle/{id}/status`. 

I've tried the `/telemetry` path but haven't managed to get it working at all : have you manage to make it work (I understand that this is a depreciated path, which might be a good reason for it not to work) ?

Unfortunately, the temperature also isn't retrieved by the `/status` path under the "environment" section which is always empty (most of the other sections work). Same question : have you tried and manage to get it to work?

I'm using it on a e-208 and I'm trying to correlate the energy consumption with the outside temperature.

Thanks in advance for any answer :smiley: ! And if you could help me get it to work would be absolutely GREAT (it's well beyond my technical skills :wink:) 

**Environment**
 - OS: Ubuntu 20.04.1 LTS (GNU/Linux 5.4.0-1026-raspi aarch64) + Node Red
 - Android app Version: 1.27.0 (probably totally irrelevant to the subject)

**Comment by flobz** [comment]
Hello,
Your welcome and thanks for your feedback.
I have done some test, it seems that those information aren't available. 
Thats a shame, it will be interesting.

> I'm using it on a e-208 and I'm trying to correlate the energy consumption with the outside temperature.
Ohh that's interesting too. You could maybe use a weather api using the position of the car? 
Will you share your work ?
I work also on energy consumption, I create a dashboard to get the consumption depending of the mean speed.
Please see [feature/dashboard](https://github.com/flobz/psa_car_controller/tree/feature/dashboard)

**Comment by Pgr73** [comment]
Hi,
Thanks for your answer. 

Yes, for sure, it's really disappointing that we can't get that information (also true for most of the telemetry information :disappointed:). I guess that is a confirmation that Peugeot is more a car manufacturer than a tech company, which is not really a surprise :wink:.

You are right:  I followed your suggestion and I'm getting the information from openweathermap.org using their API. It's obviously not as precise, but should be a good enough starting point. 
I guess another option would be to capture data directly from the CAN bus through the car's OBD plug ..... but that is not straightforward at all as the protocol is not standard on EVs :sob: ! Maybe for a future version !

Sure, I will get to sharing what I have done, but as I'm very new to Github and I will need a bit of time to get my project(s) "Github-compatible" (such as separating credentials, tidying my code ....) :smirk:

Yes, I did have a look at your dashboard feature code but did not implement it : I got my car a few month ago and I started on my project before your repo was published....and I followed a different route to meet what was my original objective. I wanted to compensate for the limited functionality provided by Peugeot in their App and control the charge to limit charging to a certain battery level (not sure this will make a big difference on the battery SOH in the long run), while maximising charge during low electricity rates during the night. 

Once you made your project available , I used it to extend the functionality and log the car's data and automate the consumption monitoring.

FYI, I'm using an ESP32 with a touchscreen to drive a charge relay (using a GreenUp plug) and my data is consolidated on a RPI 4 using MQTT, NodeRed and MySQL and displayed using Grafana.

Rgds
